### PR TITLE
Handle edge case caused by non-standard USFM

### LINKED ISSUE #520: Error running analyse_project_pairs
When I try to run analyze_project_pairs I get the following error:
```
F:\GitHub\silnlp>poetry run python -m silnlp.nmt.analyze_project_pairs FT-Dizin/align  
2024-09-16 16:52:45,571 - silnlp.common.environment - INFO - Using workspace: S:/ as per environment variable SIL_NLP_DATA_PATH.
2024-09-16 16:52:51,858 - silnlp.common.utils - INFO - Git commit: 2eb2026b39
2024-09-16 16:52:53,135 - silnlp.nmt.clearml_connection - INFO - No ClearML task initiated.
<...>
2024-09-16 16:56:54,575 - silnlp.common.collect_verse_counts - WARNING - en-NRSV contains text in books ['TOB', 'JDT', 'ESG', 'WIS', 'SIR', 'BAR', 'LJE', 'S3Y', 'SUS', 'BEL', '1MA', '2MA', '3MA', '4MA', '1ES', '2ES', 'MAN']. Use --deutero to include counts for these books.
Traceback (most recent call last):
  File "C:\Python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "F:\GitHub\silnlp\silnlp\nmt\analyze_project_pairs.py", line 519, in <module>
    main()
  File "F:\GitHub\silnlp\silnlp\nmt\analyze_project_pairs.py", line 501, in main
    get_corpus_stats(config, args.recalculate, args.deutero)
  File "F:\GitHub\silnlp\silnlp\nmt\analyze_project_pairs.py", line 67, in get_corpus_stats
    corpus = get_scripture_parallel_corpus(src_file.path, trg_file.path, False)
  File "F:\GitHub\silnlp\silnlp\common\corpus.py", line 76, in get_scripture_parallel_corpus
    vrefs[-1] = VerseRef.from_range(vrefs[-1], vref)
  File "C:\Users\David\AppData\Local\pypoetry\Cache\virtualenvs\silnlp-X71j0jgb-py3.8\lib\site-packages\machine\scripture\verse_ref.py", line 99, in from_range      
    raise ValueError("The start and end verses are not in the same chapter.")
ValueError: The start and end verses are not in the same chapter.
```

I modify the classmethod from_range in machine/scripture/verse_ref.py to include more details:
```
    @classmethod
    def from_range(cls, start: VerseRef, end: VerseRef) -> VerseRef:
        #print(f"start: {start}  end: {end}")
        if start.book_num != end.book_num or start.chapter_num != end.chapter_num:
            raise ValueError(f"The start {start} and end {end} verses are not in the same book or chapter.")
        if start.verse_num == end.verse_num:
            raise ValueError(f"The start and end verse numbers are the same: {start}")
        if start.has_multiple:
            raise ValueError(f"This start verse contains multiple verses. {start}")
        if end.has_multiple:
            raise ValueError(f"This end verse contains multiple verses. {end}")
        if start.versification != end.versification:
            raise ValueError(f"The start {start} and end {end}verses do not have the same versification.")	
```

Then the error has a bit more info:
```ValueError: The start WIS 19:22 and end SIR 19:1 verses are not in the same book or chapter.```

### PR DESCRIPTION
**Pull request by isaac091**
TL;DR: A project with non-standard USFM was causing `get_scripture_parallel_corpus` to try to create illegal vref ranges. I don't think this particular scenario could happen with normal USFM, but the change won't affect standard use at all.

This error came about from a project with very non-standard USFM (the files are `.PTW`, but if that is another standard, I can't imagine that these files followed that either). In the files I looked at, there were lots of missing verse markers, chapter markers, etc.. This resulted in an extracted corpus with lots of `<range>` markers. In one case, a whole book was entirely `<range>` markers, so `get_scripture_parallel_corpus` was trying to create invalid cross-book VerseRef ranges, and it only got caught when the chapter number looped back around to that of the first vref.

Closes #520

- - -
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/silnlp/522)

### CODE REVIEW

**Review by ddaspit — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/silnlp/522#-:-O6wNNAg5j6x1QvKKsIS:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @isaac091)
